### ISSUE TITLE
[ACD + FUSE] panic

### ISSUE THREAD

**Issue by dmp42** [rclone/rclone#932]
> What is your rclone version (eg output from `rclone -V`)

rclone v1.34-DEV

On master (cb9f1eefd24e9a4d15cdee4e27396b4d773a00a9).

> Which OS you are using and how many bits (eg Windows 7, 64 bit)

macOS Sierra 10.12.1 (16B2555)

> Which cloud storage system are you using? (eg Google Drive)

Amazon Cloud Drive

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone mount acd: /Volumes/Local`

Then messing-up with `bup` on /Volumes/Local

> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)

Not too sure if this is a fuse or rclone issue.

```
2016/12/06 19:59:33 bup-test/config.lock: Attempting to move to "bup-test/config"
2016/12/06 19:59:33 : Dir.Attr
2016/12/06 19:59:33 : Dir.ReadDirAll
2016/12/06 19:59:33 : Dir.ReadDirAll OK with 7 entries
2016/12/06 19:59:33 : Dir.Attr
2016/12/06 19:59:33 : Dir.ReadDirAll
2016/12/06 19:59:33 : Dir.ReadDirAll OK with 7 entries
2016/12/06 19:59:33 bup-test/config: Dir.Rename renamed from "bup-test/config.lock"
2016/12/06 19:59:33 bup-test: Dir.Attr
2016/12/06 19:59:33 bup-test/config.lock: Dir.Lookup
2016/12/06 19:59:33 bup-test/config: Dir.Lookup
2016/12/06 19:59:33 bup-test/config: Dir.Lookup OK
2016/12/06 19:59:33 bup-test/config: File.Attr
2016/12/06 19:59:33 fuse: panic in handler for Lookup [ID=0x13 Node=0x4 Uid=503 Gid=20 Pid=68458] "config": runtime error: invalid memory address or nil pointer dereference
goroutine 2199 [running]:
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve.func2(0xbd6e00, 0xc427566dc0, 0xc420214ec8, 0xc420214e1f)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:857 +0x1d9
panic(0x729fe0, 0xc4200140e0)
	/Users/dmp/Applications/bin/homebrew/Cellar/go/1.7.3/libexec/src/runtime/panic.go:458 +0x243
github.com/ncw/rclone/amazonclouddrive.(*Object).Size(0xc420469400, 0xc420469400)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/amazonclouddrive/amazonclouddrive.go:830 +0x9
github.com/ncw/rclone/cmd/mount.(*File).Attr(0xc42758e2c0, 0x26023b0, 0xc42758e280, 0xc4202ab2d8, 0x0, 0x0)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/cmd/mount/file.go:56 +0x12b
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.nodeAttr(0x26023b0, 0xc42758e280, 0xbcf380, 0xc42758e2c0, 0xc4202ab2d8, 0xc42046f750, 0x6)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:260 +0x11a
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).saveLookup(0xc4202aa000, 0x26023b0, 0xc42758e280, 0xc4202ab2c0, 0xc4204b4400, 0xc42046f750, 0x6, 0xbcf380, 0xc42758e2c0, 0x200, ...)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1404 +0x63
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).handleRequest(0xc4202aa000, 0x26023b0, 0xc42758e280, 0xbcf300, 0xc420020420, 0xc4204b4400, 0xbd6e00, 0xc427566dc0, 0xc420214ec8, 0x0, ...)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1057 +0x3d6f
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve(0xc4202aa000, 0xbd6e00, 0xc427566dc0)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:878 +0x4a3
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve.func1(0xc4202aa000, 0xbd6e00, 0xc427566dc0)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:425 +0x6e
created by github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:426 +0x3e4
2016/12/06 19:59:33 bup-test/config: Dir.Lookup
2016/12/06 19:59:33 bup-test/config: Dir.Lookup OK
2016/12/06 19:59:33 bup-test/config: File.Attr
2016/12/06 19:59:33 fuse: panic in handler for Lookup [ID=0x25 Node=0x4 Uid=503 Gid=20 Pid=68458] "config": runtime error: invalid memory address or nil pointer dereference
goroutine 2218 [running]:
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve.func2(0xbd6e00, 0xc427567280, 0xc420214ec8, 0xc420214e1f)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:857 +0x1d9
panic(0x729fe0, 0xc4200140e0)
	/Users/dmp/Applications/bin/homebrew/Cellar/go/1.7.3/libexec/src/runtime/panic.go:458 +0x243
github.com/ncw/rclone/amazonclouddrive.(*Object).Size(0xc420469400, 0xc420469400)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/amazonclouddrive/amazonclouddrive.go:830 +0x9
github.com/ncw/rclone/cmd/mount.(*File).Attr(0xc42758e2c0, 0x26023b0, 0xc42758e4c0, 0xc4202ab518, 0x0, 0x0)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/cmd/mount/file.go:56 +0x12b
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.nodeAttr(0x26023b0, 0xc42758e4c0, 0xbcf380, 0xc42758e2c0, 0xc4202ab518, 0xc42046f756, 0x6)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:260 +0x11a
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).saveLookup(0xc4202aa000, 0x26023b0, 0xc42758e4c0, 0xc4202ab500, 0xc4204b4400, 0xc42046f756, 0x6, 0xbcf380, 0xc42758e2c0, 0x200, ...)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1404 +0x63
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).handleRequest(0xc4202aa000, 0x26023b0, 0xc42758e4c0, 0xbcf300, 0xc420020420, 0xc4204b4400, 0xbd6e00, 0xc427567280, 0xc420214ec8, 0x0, ...)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:1057 +0x3d6f
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).serve(0xc4202aa000, 0xbd6e00, 0xc427567280)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:878 +0x4a3
github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve.func1(0xc4202aa000, 0xbd6e00, 0xc427567280)
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:425 +0x6e
created by github.com/ncw/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve
	/Users/dmp/dev/go/src/github.com/ncw/rclone/vendor/bazil.org/fuse/fs/serve.go:426 +0x3e4
2016/12/06 19:59:33 bup-test: Dir.Lookup
2016/12/06 19:59:33 bup-test: Dir.Lookup OK
2016/12/06 19:59:33 bup-test: Dir.Attr
...
```

**Comment by dmp42** [comment]
So, `o.info` is nil. Trying to force a `o.readMetadata` just before ends-up with:

```
2016/12/06 20:25:45 bup-test/config: Dir.Rename renamed from "bup-test/config.lock"
2016/12/06 20:25:45 bup-test/config: Failed to read metadata: object not found
2016/12/06 20:25:46 bup-test/config: Failed to read metadata: object not found
2016/12/06 20:25:48 bup-test/config.lock: Dir.Rename error: can't move object - incompatible remotes
```

**Comment by ncw** [comment]
How do you reproduce this?

I'm guessing something to do with moves - in which case try

http://beta.rclone.org/v1.34-60-g2656a0e/ (uploaded in 15-30 mins) or just update your git repo if you prefer.  I've re-worked the move code.

If it still doesn't work, then can you make a repro for me so I can make it go wrong here.

Thanks

**Comment by dmp42** [comment]
Thanks a lot. I'll give it a try and report here.
I'll also try to get a reproducible/reasonably small test case.

**Comment by ncw** [comment]
I think this is very likely fixed by #973 

You can try this here

http://beta.rclone.org/v1.35-26-gaa62e93/ (uploaded in 15-30 mins)

**Comment by animosity22** [comment]
Legacy ACD issue, closing out.
